SimCalorimetry/EcalSimAlgos/interface/EcalSignalGenerator.h

```diff
@@ -81,6 +92,45 @@ class EcalSignalGenerator : public EcalBaseSignalGenerator {
     m_maxEneEB = (agc->getEBValue()) * theDefaultGains[1] * MAXADC * m_EBs25notCont;
     m_maxEneEE = (agc->getEEValue()) * theDefaultGains[1] * MAXADC * m_EEs25notCont;
 
+    if (m_timeDependent) {
+      //----
+      // Ecal LaserCorrection Constants for laser correction ratio
+      edm::ESHandle<EcalLaserDbService> laser;
+      eventSetup->get<EcalLaserDbRecord>().get(laser);
+
+      //
+      // FIXME: is this workaround of using "run" really needed or "time" can be used in MC generation as well?
+      //        check with generation experts
+      //
+      const edm::TimeValue_t eventTimeValue = theEvent->run();
```

Comment:
I don't understand this. `Event::run()` returns a `RunNumber_t`: https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/interface/Event.h#L101. The `RunNumber_t` is in turn just an `unsigned int`: https://github.com/cms-sw/cmssw/blob/11d0e88b57a38348e6118772ba1bda6a573c6a00/DataFormats/Provenance/interface/RunLumiEventNumber.h#L14 In contrast, [TimeValue_t](https://github.com/cms-sw/cmssw/blob/9308c4dcd88414e6d55069443c42df1121a0175d/DataFormats/Provenance/interface/Timestamp.h) encodes the number of seconds and microseconds since the start of January 1, 1970. I think that you intend to use:
```
const edm::TimeValue_t eventTimeValue = theEvent->runAuxiliary().beginTime().value();
```
Ideally, one would use the following:
```
const edm::TimeValue_t eventTimeValue = theEvent->getLuminosityBlock().beginTime().value();
```
but that's currently not possible because the lumi-section values are currently used in encode some technical information about the MC production.

Reply:
Correct. I'm going to fix this now.

Reply:
Actually "runAuxiliary()" does not seem to be a method of the class "edm::Event"

Is there another way to access to the time of the run?

Reply:
Sorry, this should have read
```
const edm::TimeValue_t eventTimeValue = theEvent->getRun().runAuxiliary().beginTime().value();
```

Reply:
Time can not be used with MC only Run, Lumi

Reply:
Thanks! 
New commit with the fix is coming ...

Reply:
> Time can not be used with MC only Run, Lumi

ops ... I was too fast.

No time is associated with a "run" in the MC then?

Reply:
> Time can not be used with MC only Run, Lumi

@Dr15Jones Is this a consequence of MC having only a trivial payload for the relevant record (RunInfoRcd?) or is this a consequence of some constraint imposed by the framework?